libminifi/include/core/state/nodes/MetricsBase.h

```diff
@@ -159,7 +161,7 @@ class ObjectNode : public ResponseNode {
 class ResponseNodeSource {
  public:
   virtual ~ResponseNodeSource() = default;
-  virtual std::shared_ptr<ResponseNode> getResponseNodes() = 0;
+  virtual SharedResponseNode getResponseNodes() = 0;
```

Comment:
can this return more than one response nodes? should we drop the `s`?

Reply:
Good point, updated in c71bdaa3378b7087a33b7bd5f9a6bf4c3dd8cff8